okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -157,18 +163,14 @@ public void transportReportStatus(Status newStatus, boolean stopDelivery,
 
   @Override
   protected void sendCancel() {
-    if (transport.finishStream(id(), Status.CANCELLED)) {
-      frameWriter.rstStream(id(), ErrorCode.CANCEL);
-      transport.stopIfNecessary();
-    }
+    frameWriter.rstStream(id(), ErrorCode.CANCEL);
```

Comment:
Note that before we would only send RST_STREAM if the stream existed (e.g., if the stream is cancelled multiple times, only the first will RST_STREAM). It seems like we would want that optimization.